### ISSUE TITLE
Patch validation fail with one of required fields

### ISSUE THREAD

**Issue by joaonice** [Vincit/objection.js#648]
Hi,

giving the following model JSON schema:

```json
{
   "oneOf": [
      { "required": ["email"] },
      { "required": ["username"] }
   ],
   "properties": {
      "email": { "type": "string" },
      "password": { "type": "string" },
      "username": { "type": "string" }
   },
   "type": "object"
}
```

If we want to **PATCH** one instance of this model only with the `password` field, we have a validation error because one of the fields (email or username) is required. Patch operations ignores the `required` fields. Shouldn't the `oneOf` required fields be ignored as well? 

Thank you!

**Comment by koskimas** [comment]
Ha, I didn't know `required` could be set like that. Are you sure that's a supported syntax? If it is then yes, those should be ignored as well

**Comment by joaonice** [comment]
This syntax is supported (example in ajv documentation https://github.com/epoberezkin/ajv#filtering-data)

**Comment by koskimas** [comment]
Ok. Thanks for bringing this up. I'll get this fixed in the next patch version.

**Comment by joaonice** [comment]
@koskimas, any plans for releasing `0.9.3` ?

**Comment by koskimas** [comment]
@joaonice I released 0.9.3 today

**Comment by joaonice** [comment]
@koskimas thanks!
